src/client/mvt/feature-decoder.js

```diff
@@ -53,26 +53,46 @@ export function isClockWise(vertices) {
     return a > 0;
 }
 
+const CLIPMAX = 1;
+const CLIPMIN = -CLIPMAX;
+
+const clippingEdges = [
+    p => p[0] <= CLIPMAX,
+    p => p[1] <= CLIPMAX,
+    p => p[0] >= CLIPMIN,
+    p => p[1] >= CLIPMIN,
+];
+
+const clippingEdgeIntersectFn = [
+    (a, b) => geometryUtils.intersect(a, b, [CLIPMAX, -100], [CLIPMAX, 100]),
+    (a, b) => geometryUtils.intersect(a, b, [-100, CLIPMAX], [100, CLIPMAX]),
+    (a, b) => geometryUtils.intersect(a, b, [CLIPMIN, -100], [CLIPMIN, 100]),
+    (a, b) => geometryUtils.intersect(a, b, [-100, CLIPMIN], [100, CLIPMIN]),
+];
+
 export function clipPolygon(preClippedVertices, polygon, isHole) {
     // Sutherland-Hodgman Algorithm to clip polygons to the tile
     // https://www.cs.drexel.edu/~david/Classes/CS430/Lectures/L-05_Polygons.6.pdf
-    const clippingEdges = [
-        p => p[0] <= 1,
-        p => p[1] <= 1,
-        p => p[0] >= -1,
-        p => p[1] >= -1,
-    ];
-
-    const clippingEdgeIntersectFn = [
-        (a, b) => geometryUtils.intersect(a, b, [1, -10], [1, 10]),
-        (a, b) => geometryUtils.intersect(a, b, [-10, 1], [10, 1]),
-        (a, b) => geometryUtils.intersect(a, b, [-1, -10], [-1, 10]),
-        (a, b) => geometryUtils.intersect(a, b, [-10, -1], [10, -1]),
-    ];
+
+    let clippedTypes = {};
 
     // for each clipping edge
     for (let i = 0; i < 4; i++) {
```

Comment:
Why 4?

Reply:
Same as before, we need to clip against the 4 edges of the tile